Makefile

```diff
@@ -925,3 +924,100 @@ ui-development: build/api build/api-db build/local-api-data-watcher-pusher build
+		&& echo '    insecure_skip_verify = true'                                                                    >> $$KINDCONFIG \
+		&& echo '  [plugins."io.containerd.grpc.v1.cri".registry.mirrors."registry.192.168.48.2.nip.io:32080"]'      >> $$KINDCONFIG \
+		&& echo '    endpoint = ["http://registry.192.168.48.2.nip.io:32080"]'                                       >> $$KINDCONFIG \
+		&& echo '  [plugins."io.containerd.grpc.v1.cri".registry.configs."registry.192.168.160.2.nip.io:32443".tls]' >> $$KINDCONFIG \
+		&& echo '    insecure_skip_verify = true'                                                                    >> $$KINDCONFIG \
+		&& echo '  [plugins."io.containerd.grpc.v1.cri".registry.mirrors."registry.192.168.160.2.nip.io:32080"]'     >> $$KINDCONFIG \
+		&& echo '    endpoint = ["http://registry.192.168.160.2.nip.io:32080"]'                                      >> $$KINDCONFIG \
+		&& echo '  [plugins."io.containerd.grpc.v1.cri".registry.configs."registry.172.25.0.2.nip.io:32443".tls]'    >> $$KINDCONFIG \
+		&& echo '    insecure_skip_verify = true'                                                                    >> $$KINDCONFIG \
+		&& echo '  [plugins."io.containerd.grpc.v1.cri".registry.mirrors."registry.172.25.0.2.nip.io:32080"]'        >> $$KINDCONFIG \
+		&& echo '    endpoint = ["http://registry.172.25.0.2.nip.io:32080"]'                                         >> $$KINDCONFIG \
+		&& echo 'nodes:'                                                                                             >> $$KINDCONFIG \
+		&& echo '- role: control-plane'                                                                              >> $$KINDCONFIG \
+		&& echo '  image: $(KIND_IMAGE)'                                                                             >> $$KINDCONFIG \
+		&& echo '  extraMounts:'                                                                                     >> $$KINDCONFIG \
+		&& echo '  - containerPath: /var/lib/kubelet/config.json'                                                    >> $$KINDCONFIG \
+		&& echo '    hostPath: $(HOME)/.docker/config.json'                                                          >> $$KINDCONFIG \
+		&& KIND_CLUSTER_NAME="$(CI_BUILD_TAG)" ./local-dev/kind create cluster --config=$$KINDCONFIG \
+		&& echo -e 'Interact with the cluster during the test run like so:\n' \
+		&& echo "export KUBECONFIG=\$$(mktemp) && scp $$NODE_NAME:$$KUBECONFIG \$$KUBECONFIG && KIND_PORT=\$$(sed -nE 's/.+server:.+:([0-9]+)/\1/p' \$$KUBECONFIG) && ssh -fNL \$$KIND_PORT:127.0.0.1:\$$KIND_PORT $$NODE_NAME" \
+		&& echo -e 'kubectl ...\n'
+
+KIND_SERVICES = api api-db api-redis auth-server broker controllerhandler drush-alias keycloak keycloak-db ssh
+
+.PHONY: kind/preload
+kind/preload: kind/cluster $(addprefix build/,$(KIND_SERVICES))
+	for image in $(KIND_SERVICES); do \
+		KIND_CLUSTER_NAME="$(CI_BUILD_TAG)" ./local-dev/kind load docker-image $(CI_BUILD_TAG)/$$image; \
```

Comment:
we decided in the k3d testing infra against the `load` feature to push images inside the kubernetes cluster, mostly because:
1. The loading takes a lot of time and also does not profit of any docker layer caching
2. if you have a tiny change on the image you need to load the whole image again, which takes a lot of time again
3. If you stop the kind cluster and start it again you need to push all images again. 

This is why we create a local-registry and push the images in there:
https://github.com/amazeeio/lagoon/blob/fix-ci/Makefile#L329-L336

and then use them inside k3d (search for `172.17.0.1:5000`)

Do you think we could use the same system also for kind? would make a bit a nicer DX

Reply:
This is an interesting point that I hadn't considered. I'll see if I can rework this a bit.

Reply:
Okay, so this now uses the installed harbor registry to push/pull images. It does seem to be faster for iterative local development in my testing. Could you try it?